lib/wallet/src/Cardano/Wallet/Address/Discovery/Random.hs

```diff
@@ -455,13 +455,10 @@ instance KnownNat p => IsOurs (RndAnyState n p) Address where
         double :: Integral a => a -> Double
         double = fromIntegral
 
-        -- ADP-3056: Consider making this implementation a part of a
-        -- @validate :: UTxOAssumptions -> Address -> Bool@ function
-        -- somewhere in the Write module hierarchy.
-        correctAddressType :: Bool
-        correctAddressType = case toLedger (Address bytes) of
-            Ledger.Addr _net _ _ -> False
-            Ledger.AddrBootstrap _ -> True
+        correctAddressType =
+            Write.addressSatisfiesAssumptions
```

Comment:
I guess this is a place where something like 
```suggestion
            UTxOAssumptions.validateAddress
```
could make sense

Reply:
Or we could have both:
- `UTxOAssumptions.validate`
- `UTxOAssumptions.validateAll` (will need to introduce later for ADP-3055)

Reply:
I would definitely be in favour of something like:
```hs
UTxOAssumptions.validateAddress address
```